### PR TITLE
Adds support for Noobaa standalone in cluster overview

### PR DESCRIPTION
**Pull request by bipuladh**
[Bug 2256563](https://bugzilla.redhat.com/show_bug.cgi?id=2256563)

Changes:
- Fixes StatusAndIconText support for HealthStates; previously it dint support for it.
- Adds text `Healthy` for OK State,
- Moves to a `console.dashboards/overview/health/prometheus` from `console.dashboards/overview/health/resource
`
 

#### Screenshot
Screenshot from 2024-04-16 21-25-12

### CODE REVIEW

File: packages/shared/src/selectors/k8s.ts [hunk 1]
```diff
@@ -51,3 +55,16 @@ export const getOwnerReferences = <
     value,
     'metadata.ownerReferences'
   ) as K8sResourceCommon['metadata']['ownerReferences'];
+
+type KnownResourceConditions = 'Available' | 'Degraded' | 'Progressing';
+
+export const getResourceCondition = <
```
Comment (bipuladh) [inline packages/shared/src/selectors/k8s.ts:61]: Not used right now but is definitely useful for future hence not removing it for now.

File: packages/odf/components/cluster-overview-extensions/HealthOverview.tsx [hunk 2]
```diff
@@ -1,24 +1,76 @@
...
+) => {
+  const isNoobaaOnly = _.isEmpty(ceph?.data);
+  if (!isNoobaaOnly) {
+    return getCephsHealthState({
+      ceph: {
+        data: ceph.data as K8sResourceCommon[],
+        loaded: ceph.loaded,
+        loadError: ceph.loadError,
+      },
+    });
```
Comment (SanjalKatiyar) [inline packages/odf/components/cluster-overview-extensions/HealthOverview.tsx:35]: if it's normal deployment (noobaa + ceph both), should we include both statuses here as well ??

File: packages/odf/components/cluster-overview-extensions/HealthOverview.tsx [hunk 3]
```diff
@@ -1,24 +1,76 @@
...
+      },
+    },
+    t
+  );
+  const healthStatus: HealthState = isNoobaaOnly
+    ? noobaaHealth?.state
+    : cephHealthState?.state;
+  const healthMessage: string = isNoobaaOnly
+    ? noobaaHealth?.message
+    : cephHealthState?.message;
```
Comment (SanjalKatiyar) [inline packages/odf/components/cluster-overview-extensions/HealthOverview.tsx:73]: same here, if we have both noobaa + ceph, should we reflect both ??

**Review by SanjalKatiyar — COMMENTED** [review]
Can we try to make it as close as possible to SS overview status-card: https://github.com/red-hat-storage/odf-console/blob/master/packages/odf/components/odf-dashboard/status-card/status-card.tsx#L68-L80 ??
so that there is no mismatch between status/health between 2 different dashboards (someone can raise a bug if there is a mismatch)...

File: packages/odf/components/cluster-overview-extensions/HealthOverview.tsx (hunk 3 above)
Comment (bipuladh) [reply packages/odf/components/cluster-overview-extensions/HealthOverview.tsx:73]: currently to fix bug I am limiting to just supporting Noobaa when Ceph is absent but we need to rework on this to support multicluster scenario properly.

File: packages/odf/components/cluster-overview-extensions/HealthOverview.tsx (hunk 2 above)
Comment (bipuladh) [reply packages/odf/components/cluster-overview-extensions/HealthOverview.tsx:35]: currently to fix bug I am limiting to just supporting Noobaa when Ceph is absent but we need to rework on this to support multicluster scenario properly.

File: packages/odf/components/cluster-overview-extensions/HealthOverview.tsx [hunk 4]
```diff
@@ -48,14 +99,11 @@ export const StoragePopover: React.FC<StoragePopoverProps> = ({ ceph }) => {
...
 
-export const healthResource = {
-  ceph: {
-    kind: referenceForModel(CephClusterModel),
-    namespaced: false,
-    isList: true,
-  },
+export const healthResource: FirehoseResource = {
+  kind: referenceForModel(CephClusterModel),
+  namespaced: true,
```
Comment (SanjalKatiyar) [inline packages/odf/components/cluster-overview-extensions/HealthOverview.tsx:106]:
```suggestion
  namespaced: false,
```

File: packages/shared/src/status/HealthStatus.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import { UnknownIcon } from '@patternfly/react-icons/dist/esm/icons/unknown-icon';
+import { useCustomTranslation } from '../useCustomTranslationHook';
+import { GreenCheckCircleIcon } from './icons';
+
+type HealthStatusWithIconProps = {
+  state: HealthState;
+  message?: string;
+};
+
+export const HealthStatusWithIcon: React.FC<HealthStatusWithIconProps> = ({
```
Comment (SanjalKatiyar) [inline packages/shared/src/status/HealthStatus.tsx:17]: `healthStateMapping` from `shared/src/dashboards/status-card/states.tsx` will not work ?

Comment (SanjalKatiyar) [reply packages/shared/src/status/HealthStatus.tsx:17]: I think we have many state mappings like these...

File: packages/shared/src/status/HealthStatus.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  switch (state) {
+    case HealthState.OK:
+      return (
+        <StatusIconAndText
+          icon={<GreenCheckCircleIcon />}
+          title={t('Healthy')}
+        />
+      );
+    case HealthState.PROGRESS:
+      return <StatusIconAndText icon={<HourglassHalfIcon />} title={message} />;
```
[the remark this answers is not in the document]
Comment (SanjalKatiyar) [reply packages/shared/src/status/HealthStatus.tsx:31]: https://github.com/red-hat-storage/odf-console/pull/1294/files#r1568408785

**Comment by SanjalKatiyar** [comment]
/retitle Adds support for Noobaa standalone in cluster overview
(will add bug number in `release-4.16` PR)

File: packages/shared/src/status/HealthStatus.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/shared/src/status/HealthStatus.tsx:31]: OCP uses that

File: packages/odf/components/cluster-overview-extensions/HealthOverview.tsx (hunk 4 above)
Comment (bipuladh) [reply packages/odf/components/cluster-overview-extensions/HealthOverview.tsx:106]: It is a namespaced resource.

File: packages/shared/src/status/HealthStatus.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (bipuladh) [reply packages/shared/src/status/HealthStatus.tsx:31]: but moving to healthstatemapping for now

File: packages/odf/components/cluster-overview-extensions/HealthOverview.tsx (hunk 4 above)
Comment (bipuladh) [reply packages/odf/components/cluster-overview-extensions/HealthOverview.tsx]: I guess it is to query all the Crs from all ns's

Comment (SanjalKatiyar) [reply packages/odf/components/cluster-overview-extensions/HealthOverview.tsx]: doesn't we set `namespaced: true` when we are passing specific `namespace` to this object as well ??

else if we want to poll CRs from multiple namespaces we pass `namespaced: false` ?? (afaik that's how we do in other places as well)

**Pull request merged**